azure/services/async/async.go

```diff
@@ -68,6 +68,36 @@ func processOngoingOperation(ctx context.Context, scope FutureScope, client Futu
 	return result, err
 }
 
+// GetExistingResource implements the logic for creating a resource.
+func GetExistingResource(ctx context.Context, scope FutureScope, client Creator, spec azure.ResourceSpecGetter, serviceName string) (interface{}, error) {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "async.Service.GetExistingResource")
+	defer done()
+
+	resourceName := spec.ResourceName()
+	rgName := spec.ResourceGroupName()
+
+	// TODO: should Get operations be blocked by a long running operation?
```

Comment:
no need to check for long running operations here, if we reach this phase it means there is already no running operation